src/services/reviews.js

```diff
@@ -0,0 +1,534 @@
+				reviewer_id: userId,
+				resource_id: resourceId,
+			})
+
+			//check review is exist
+			if (!reviewResource || !reviewResource.id) {
+				return responses.failureResponse({
+					message: 'REVIEW_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			const review = await reviewsQueries.findOne({
+				organization_id: reviewResource.organization_id,
+				resource_id: resourceId,
+				reviewer_id: userId,
+			})
+
+			if (!review || !review.id) {
+				return responses.failureResponse({
+					message: 'REVIEW_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			//check already someone started review
+			if (review.status === common.REVIEW_STATUS_NOT_STARTED) {
+				const existingReview = await reviewsQueries.findOne({
+					organization_id: resource.organization_id,
+					resource_id: resourceId,
+					reviewer_id: { [Op.notIn]: [userId] },
+					status: { [Op.in]: _notAllowedReviewStatus },
+				})
+
+				if (existingReview?.id) {
+					return responses.failureResponse({
+						message: 'REVIEW_INPROGRESS',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+					})
+				}
+			}
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'VALIDATION_PASSED',
+				result: {
+					resource,
+					review,
+				},
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Approve the review
+	 * @method
+	 * @name handleApproval
+	 * @param {Integer} reviewId - review id.
+	 * @param {Object} bodyData - review body data.
+	 * @param {Integer} resourceId - resource id.
+	 * @param {String} loggedInUserId - logged in user id.
+	 * @param {Array} comments - comments.
+	 * @param {String} minApproval - min approval
+	 * @param {String} reviewOrgId - review org id.
+	 * @returns {JSON} - review updated response.
+	 */
+	static async handleApproval(reviewId, resourceId, loggedInUserId, comments = [], minApproval, reviewOrgId) {
+		try {
+			// handle approval of a review
+			let publishResource = false
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			await reviewsQueries.update(
+				{ id: reviewId, organization_id: reviewOrgId },
+				{ status: common.RESOURCE_STATUS_APPROVED }
+			)
+
+			// Check if the number of approved reviews meets or exceeds the minimum required approvals
+			const reviewsApproved = await reviewsQueries.count({
+				resource_id: resourceId,
+				status: common.REVIEW_STATUS_APPROVED,
+			})
+
+			if (minApproval <= reviewsApproved) {
+				publishResource = true
+			}
+
+			return publishResource
+		} catch (error) {
+			throw error
+		}
+	}
+}
+
+const _notAllowedStatusForReview = [
+	common.RESOURCE_STATUS_REJECTED,
+	common.RESOURCE_STATUS_REJECTED_AND_REPORTED,
+	common.RESOURCE_STATUS_PUBLISHED,
+]
+
+const _notAllowedReviewStatus = [
```

Comment:
@priyanka-TL Please add comments on what this list is and where its used. The name is not very clear, consider changing it.